init-tools.sh

```diff
@@ -65,55 +61,59 @@ execute_with_retry() {
     return 0
 }
 
+__PKG_ARCH=x64
+if [ "$(uname -m)" == "aarch64" ]; then
+    __PKG_ARCH=arm64
+fi
+
+OSName=$(uname -s)
+case $OSName in
+    Darwin)
+        OS=OSX
+        __PKG_RID=osx
+        ulimit -n 2048
+        # Format x.y.z as single integer with three digits for each part
+        VERSION=`sw_vers -productVersion| sed -e 's/\./ /g' | xargs printf "%03d%03d%03d"`
+        if [ "$VERSION" -lt 010012000 ]; then
+            echo error: macOS version `sw_vers -productVersion` is too old. 10.12 is needed as minimum.
+            exit 1
+        fi
+        ;;
+
+    Linux)
+        __PKG_RID=linux
+        OS=Linux
+
+        if [ -e /etc/os-release ]; then
+            source /etc/os-release
+            if [[ $ID == "alpine" ]]; then
+                # remove the last version digit
+                VERSION_ID=${VERSION_ID%.*}
+            __PKG_RID=alpine.$VERSION_ID
+            fi
+        elif [ -e /etc/redhat-release ]; then
+            redhatRelease=$(</etc/redhat-release)
+            if [[ $redhatRelease == "CentOS release 6."* || $redhatRelease == "Red Hat Enterprise Linux Server release 6."* ]]; then
+            __PKG_RID=rhel.6
+            fi
+        fi
+
+        ;;
+
+    *)
+    echo "Unsupported OS '$OSName' detected. Downloading linux-$__PKG_ARCH tools."
+        OS=Linux
+        __PKG_RID=linux
+        ;;
+esac
+
```

Comment:
Code moved out of https://github.com/dotnet/coreclr/pull/16709/files#diff-c02be70df3e835d946f675bd0bedc234R110 block below and indentation corrected.

We need to correctly set `__PKG_RID` so that we can fetch the correct `ILASM` when we initialize buildtools below.